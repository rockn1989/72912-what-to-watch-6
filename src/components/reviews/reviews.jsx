import React from 'react';
import Review from '../review/review';
import propTypes from 'prop-types';

const Reviews = ({reviews}) => {

  const [...rows] = [...[reviews.slice(0, Math.ceil(reviews.length / 2))], ...[reviews.slice(Math.ceil(reviews.length / 2))]];

  const post = rows.map((columns, idx) => {
    return (
      <div key={`${columns}_${idx}`} className="movie-card__reviews-col">
        {columns.map((review) => {
          return <Review key={`${review.date}_${idx}`} {...review} />;
        })}
      </div>
    );
  });

  return (
    <div className="movie-card__reviews movie-card__row">
      {post}
    </div>
  );
};

Reviews.propTypes = {
  reviews: propTypes.array.isRequired,
};

export default Reviews;
