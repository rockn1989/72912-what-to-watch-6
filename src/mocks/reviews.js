export default [
  {
    id: 1,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    rating: 8.9,
    date: `December 24, 2018`,
  },
  {
    id: 2,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Alex Kim`,
    rating: 5.9,
    date: `November 13, 2017`,
  },
  {
    id: 3,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Gun gun`,
    rating: 8.2,
    date: `August 4, 2018`,
  },
  {
    id: 4,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Tomy Li Jons`,
    rating: 6.6,
    date: `Jule 16, 2018`,
  },
  {
    id: 5,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Timmy`,
    rating: 5.5,
    date: `September 3, 2018`,
  },
];
