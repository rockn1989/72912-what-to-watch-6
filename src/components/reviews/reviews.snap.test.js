import React from 'react';
import {render} from '@testing-library/react';
import Reviews from './reviews';

import reviews from '../../mocks/reviews';

it(`Should Review render correctly`, () => {
  const {container} = render(<Reviews reviews={reviews} />);
  expect(container).toMatchSnapshot();
});
