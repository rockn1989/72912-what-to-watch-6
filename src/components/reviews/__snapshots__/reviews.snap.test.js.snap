// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Review render correctly 1`] = `
<div>
  <div
    class="movie-card__reviews movie-card__row"
  >
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Kate Muir
            </cite>
            <time
              class="review__date"
              dateime="2016-12-24"
            >
              December 24, 2018
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          8.9
        </div>
      </div>
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Alex Kim
            </cite>
            <time
              class="review__date"
              dateime="2016-12-24"
            >
              November 13, 2017
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          5.9
        </div>
      </div>
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Gun gun
            </cite>
            <time
              class="review__date"
              dateime="2016-12-24"
            >
              August 4, 2018
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          8.2
        </div>
      </div>
    </div>
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Tomy Li Jons
            </cite>
            <time
              class="review__date"
              dateime="2016-12-24"
            >
              Jule 16, 2018
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          6.6
        </div>
      </div>
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Timmy
            </cite>
            <time
              class="review__date"
              dateime="2016-12-24"
            >
              September 3, 2018
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          5.5
        </div>
      </div>
    </div>
  </div>
</div>
`;
